src/Monitor/Monitor/Alerts/NewAzureRmMetricAlertRuleV2CriteriaCommand.cs

```diff
@@ -25,12 +25,25 @@ namespace Microsoft.Azure.Commands.Insights.Alerts
     /// <summary>
     /// Create Metric Criteria
     /// </summary>
-    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MetricAlertRuleV2Criteria", DefaultParameterSetName = StaticThresholdParameterSet), OutputType(typeof(PSMetricCriteria))]
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MetricAlertRuleV2Criteria", DefaultParameterSetName = StaticThresholdParameterSet), OutputType(typeof(IPSMultiMetricCriteria))]
     public class NewAzureRmMetricAlertRuleV2CriteriaCommand : MonitorCmdletBase
     {
         const string StaticThresholdParameterSet = "StaticThresholdParameterSet";
         const string DynamicThresholdParameterSet = "DynamicThresholdParameterSet";
 
+        // Create a script for the Operator parameter completer based on the active Parameter Set 
+        // Since Parameter Set isn't accesible - check if one of the dynamic threshold parameters are assigned
+        const string IsDynamicThresholdParameterSetSelectedScript = "$fakeBoundParameters.DynamicThreshold -or $fakeBoundParameters.ThresholdSensitivity -ne $null -or $fakeBoundParameters.NumberOfViolations -ne $null -or $fakeBoundParameters.NumberOfExaminedAggregatedPoints -ne $null -or $fakeBoundParameters.IgnoreDataBefore -ne $null";
+        const string DynamicOperatorValues = "'GreaterThan', 'LessThan', 'GreaterOrLessThan'";
+        const string StaticOperatorValues = "'GreaterThan', 'GreaterThanOrEqual', 'LessThan', 'LessThanOrEqual'";
+        const string OperatorCompleterScript = "param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameters)\n if (" + IsDynamicThresholdParameterSetSelectedScript + ") {" + DynamicOperatorValues + " } else {" + StaticOperatorValues + "}";
+
+        /// <summary>
+        /// Gets or sets the rule DynamicThreshold
+        /// </summary>
+        [Parameter(ParameterSetName = DynamicThresholdParameterSet, Mandatory = true, HelpMessage = "Switch parameter for using Dynamic Threshold Type")]
+        public SwitchParameter DynamicThreshold { get; set; }
```

Comment:
Breaking Change: Replace “string’ parameter DynamicThreshold with SwitchParameter DynamicThreshold – these types are incompatible, and this would break any scripts using that parameter